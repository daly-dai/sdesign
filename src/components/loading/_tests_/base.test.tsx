import { render } from '@testing-library/react';
import React from 'react';
import { describe, expect, it } from 'vitest';

import SLoading from '../index';

describe('SLoading', () => {
  it('should render correctly', () => {
    const { container } = render(<SLoading />);
    expect(container).toMatchSnapshot();
  });
});
