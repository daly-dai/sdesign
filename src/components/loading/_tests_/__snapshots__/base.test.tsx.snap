// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SLoading > should render correctly 1`] = `
<div>
  <div
    class="acss-l1tuuh"
  >
    <span
      class="anticon acss-1tt9ejx"
      role="img"
    >
      <svg
        aria-hidden="true"
        class=""
        fill="currentColor"
        focusable="false"
        height="1em"
        version="1.1"
        viewBox="0 0 500 500"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <defs>
          <lineargradient
            id="linearGradient-1"
            x1="5%"
            x2="66%"
            y1="50%"
            y2="50%"
          >
            <stop
              offset="0%"
              stop-color="#00C85F"
            />
            <stop
              offset="100%"
              stop-color="#009EFF"
            />
          </lineargradient>
          <lineargradient
            id="linearGradient-2"
            x1="40.778227%"
            x2="100%"
            y1="50%"
            y2="50%"
          >
            <stop
              offset="0%"
              stop-color="#009EFF"
            />
            <stop
              offset="100%"
              stop-color="#00C85F"
            />
          </lineargradient>
        </defs>
        <g
          fill="none"
          fill-rule="evenodd"
          stroke="none"
        >
          <path
            d="M0,207.386787 C84.3476911,207.386787 161.410264,143.703372 250,143.703372 C338.589736,143.703372 416.030714,207.386787 500,207.386787"
            id="路径-2"
            stroke="url(#linearGradient-1)"
            stroke-dasharray="672"
            stroke-dashoffset="672"
            stroke-width="38"
          >
            <animate
              attributeName="stroke-dashoffset"
              begin="0.2s"
              dur="1.6s"
              repeatCount="indefinite"
              values="672;0;-672"
            />
          </path>
        </g>
        <g
          fill="none"
          fill-rule="evenodd"
          id="画板"
          stroke="none"
          stroke-width="1"
        >
          <path
            d="M-0.00926722543,329.738885 C84.3384239,329.738885 161.400996,266.055471 249.990733,266.055471 C338.580469,266.055471 416.021447,329.738885 499.990733,329.738885"
            id="路径-2备份"
            stroke="url(#linearGradient-2)"
            stroke-dasharray="672"
            stroke-width="1.6"
            transform="translate(249.990733, 297.897178) scale(1, -1) translate(-249.990733, -297.897178) "
          >
            <animate
              attributeName="stroke-dashoffset"
              dur="1.6s"
              repeatCount="indefinite"
              values="-672;0;672"
            />
          </path>
          <path
            d="M-0.00926722543,334.738885 C84.3384239,334.738885 161.400996,271.055471 249.990733,271.055471 C338.580469,271.055471 416.021447,334.738885 499.990733,334.738885"
            id="路径-2备份-2"
            stroke="url(#linearGradient-2)"
            stroke-dasharray="672"
            stroke-width="1.6"
            transform="translate(249.990733, 302.897178) scale(1, -1) translate(-249.990733, -302.897178) "
          >
            <animate
              attributeName="stroke-dashoffset"
              dur="1.6s"
              repeatCount="indefinite"
              values="-672;0;672"
            />
          </path>
          <path
            d="M-0.00926722543,339.738885 C84.3384239,339.738885 161.400996,276.055471 249.990733,276.055471 C338.580469,276.055471 416.021447,339.738885 499.990733,339.738885"
            id="路径-2备份-3"
            stroke="url(#linearGradient-2)"
            stroke-dasharray="672"
            stroke-width="1.6"
            transform="translate(249.990733, 307.897178) scale(1, -1) translate(-249.990733, -307.897178) "
          >
            <animate
              attributeName="stroke-dashoffset"
              dur="1.6s"
              repeatCount="indefinite"
              values="-672;0;672"
            />
          </path>
          <path
            d="M-0.00926722543,344.738885 C84.3384239,344.738885 161.400996,281.055471 249.990733,281.055471 C338.580469,281.055471 416.021447,344.738885 499.990733,344.738885"
            id="路径-2备份-4"
            stroke="url(#linearGradient-2)"
            stroke-dasharray="672"
            stroke-width="1.6"
            transform="translate(249.990733, 312.897178) scale(1, -1) translate(-249.990733, -312.897178) "
          >
            <animate
              attributeName="stroke-dashoffset"
              dur="1.6s"
              repeatCount="indefinite"
              values="-672;0;672"
            />
          </path>
          <path
            d="M-0.00926722543,349.738885 C84.3384239,349.738885 161.400996,286.055471 249.990733,286.055471 C338.580469,286.055471 416.021447,349.738885 499.990733,349.738885"
            id="路径-2备份-5"
            stroke="url(#linearGradient-2)"
            stroke-dasharray="672"
            stroke-width="1.6"
            transform="translate(249.990733, 317.897178) scale(1, -1) translate(-249.990733, -317.897178) "
          >
            <animate
              attributeName="stroke-dashoffset"
              dur="1.6s"
              repeatCount="indefinite"
              values="-672;0;672"
            />
          </path>
          <path
            d="M-0.00926722543,354.738885 C84.3384239,354.738885 161.400996,291.055471 249.990733,291.055471 C338.580469,291.055471 416.021447,354.738885 499.990733,354.738885"
            id="路径-2备份-6"
            stroke="url(#linearGradient-2)"
            stroke-dasharray="672"
            stroke-width="1.6"
            transform="translate(249.990733, 322.897178) scale(1, -1) translate(-249.990733, -322.897178) "
          >
            <animate
              attributeName="stroke-dashoffset"
              dur="1.6s"
              repeatCount="indefinite"
              values="-672;0;672"
            />
          </path>
          <path
            d="M-0.00926722543,359.738885 C84.3384239,359.738885 161.400996,296.055471 249.990733,296.055471 C338.580469,296.055471 416.021447,359.738885 499.990733,359.738885"
            id="路径-2备份-7"
            stroke="url(#linearGradient-2)"
            stroke-dasharray="672"
            stroke-width="1.6"
            transform="translate(249.990733, 327.897178) scale(1, -1) translate(-249.990733, -327.897178) "
          >
            <animate
              attributeName="stroke-dashoffset"
              dur="1.6s"
              repeatCount="indefinite"
              values="-672;0;672"
            />
          </path>
          <path
            d="M-0.00926722543,364.738885 C84.3384239,364.738885 161.400996,301.055471 249.990733,301.055471 C338.580469,301.055471 416.021447,364.738885 499.990733,364.738885"
            id="路径-2备份-8"
            stroke="url(#linearGradient-2)"
            stroke-dasharray="672"
            stroke-width="1.6"
            transform="translate(249.990733, 332.897178) scale(1, -1) translate(-249.990733, -332.897178) "
          >
            <animate
              attributeName="stroke-dashoffset"
              dur="1.6s"
              repeatCount="indefinite"
              values="-672;0;672"
            />
          </path>
          <path
            d="M-0.00926722543,369.738885 C84.3384239,369.738885 161.400996,306.055471 249.990733,306.055471 C338.580469,306.055471 416.021447,369.738885 499.990733,369.738885"
            id="路径-2备份-9"
            stroke="url(#linearGradient-2)"
            stroke-dasharray="672"
            stroke-width="1.6"
            transform="translate(249.990733, 337.897178) scale(1, -1) translate(-249.990733, -337.897178) "
          >
            <animate
              attributeName="stroke-dashoffset"
              dur="1.6s"
              repeatCount="indefinite"
              values="-672;0;672"
            />
          </path>
          <rect
            fill="#049FF7"
            fill-rule="nonzero"
            height="18.5394529"
            id="矩形"
            opacity="1"
            width="147.16999"
            x="176.415005"
            y="238.306353"
          />
        </g>
      </svg>
    </span>
  </div>
</div>
`;

exports[`SLoading should render correctly 1`] = `
<div>
  <div
    class="acss-l1tuuh"
  >
    <span
      class="anticon acss-1tt9ejx"
      role="img"
    >
      <svg
        aria-hidden="true"
        class=""
        fill="currentColor"
        focusable="false"
        height="1em"
        version="1.1"
        viewBox="0 0 500 500"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <defs>
          <lineargradient
            id="linearGradient-1"
            x1="5%"
            x2="66%"
            y1="50%"
            y2="50%"
          >
            <stop
              offset="0%"
              stop-color="#00C85F"
            />
            <stop
              offset="100%"
              stop-color="#009EFF"
            />
          </lineargradient>
          <lineargradient
            id="linearGradient-2"
            x1="40.778227%"
            x2="100%"
            y1="50%"
            y2="50%"
          >
            <stop
              offset="0%"
              stop-color="#009EFF"
            />
            <stop
              offset="100%"
              stop-color="#00C85F"
            />
          </lineargradient>
        </defs>
        <g
          fill="none"
          fill-rule="evenodd"
          stroke="none"
        >
          <path
            d="M0,207.386787 C84.3476911,207.386787 161.410264,143.703372 250,143.703372 C338.589736,143.703372 416.030714,207.386787 500,207.386787"
            id="路径-2"
            stroke="url(#linearGradient-1)"
            stroke-dasharray="672"
            stroke-dashoffset="672"
            stroke-width="38"
          >
            <animate
              attributeName="stroke-dashoffset"
              begin="0.2s"
              dur="1.6s"
              repeatCount="indefinite"
              values="672;0;-672"
            />
          </path>
        </g>
        <g
          fill="none"
          fill-rule="evenodd"
          id="画板"
          stroke="none"
          stroke-width="1"
        >
          <path
            d="M-0.00926722543,329.738885 C84.3384239,329.738885 161.400996,266.055471 249.990733,266.055471 C338.580469,266.055471 416.021447,329.738885 499.990733,329.738885"
            id="路径-2备份"
            stroke="url(#linearGradient-2)"
            stroke-dasharray="672"
            stroke-width="1.6"
            transform="translate(249.990733, 297.897178) scale(1, -1) translate(-249.990733, -297.897178) "
          >
            <animate
              attributeName="stroke-dashoffset"
              dur="1.6s"
              repeatCount="indefinite"
              values="-672;0;672"
            />
          </path>
          <path
            d="M-0.00926722543,334.738885 C84.3384239,334.738885 161.400996,271.055471 249.990733,271.055471 C338.580469,271.055471 416.021447,334.738885 499.990733,334.738885"
            id="路径-2备份-2"
            stroke="url(#linearGradient-2)"
            stroke-dasharray="672"
            stroke-width="1.6"
            transform="translate(249.990733, 302.897178) scale(1, -1) translate(-249.990733, -302.897178) "
          >
            <animate
              attributeName="stroke-dashoffset"
              dur="1.6s"
              repeatCount="indefinite"
              values="-672;0;672"
            />
          </path>
          <path
            d="M-0.00926722543,339.738885 C84.3384239,339.738885 161.400996,276.055471 249.990733,276.055471 C338.580469,276.055471 416.021447,339.738885 499.990733,339.738885"
            id="路径-2备份-3"
            stroke="url(#linearGradient-2)"
            stroke-dasharray="672"
            stroke-width="1.6"
            transform="translate(249.990733, 307.897178) scale(1, -1) translate(-249.990733, -307.897178) "
          >
            <animate
              attributeName="stroke-dashoffset"
              dur="1.6s"
              repeatCount="indefinite"
              values="-672;0;672"
            />
          </path>
          <path
            d="M-0.00926722543,344.738885 C84.3384239,344.738885 161.400996,281.055471 249.990733,281.055471 C338.580469,281.055471 416.021447,344.738885 499.990733,344.738885"
            id="路径-2备份-4"
            stroke="url(#linearGradient-2)"
            stroke-dasharray="672"
            stroke-width="1.6"
            transform="translate(249.990733, 312.897178) scale(1, -1) translate(-249.990733, -312.897178) "
          >
            <animate
              attributeName="stroke-dashoffset"
              dur="1.6s"
              repeatCount="indefinite"
              values="-672;0;672"
            />
          </path>
          <path
            d="M-0.00926722543,349.738885 C84.3384239,349.738885 161.400996,286.055471 249.990733,286.055471 C338.580469,286.055471 416.021447,349.738885 499.990733,349.738885"
            id="路径-2备份-5"
            stroke="url(#linearGradient-2)"
            stroke-dasharray="672"
            stroke-width="1.6"
            transform="translate(249.990733, 317.897178) scale(1, -1) translate(-249.990733, -317.897178) "
          >
            <animate
              attributeName="stroke-dashoffset"
              dur="1.6s"
              repeatCount="indefinite"
              values="-672;0;672"
            />
          </path>
          <path
            d="M-0.00926722543,354.738885 C84.3384239,354.738885 161.400996,291.055471 249.990733,291.055471 C338.580469,291.055471 416.021447,354.738885 499.990733,354.738885"
            id="路径-2备份-6"
            stroke="url(#linearGradient-2)"
            stroke-dasharray="672"
            stroke-width="1.6"
            transform="translate(249.990733, 322.897178) scale(1, -1) translate(-249.990733, -322.897178) "
          >
            <animate
              attributeName="stroke-dashoffset"
              dur="1.6s"
              repeatCount="indefinite"
              values="-672;0;672"
            />
          </path>
          <path
            d="M-0.00926722543,359.738885 C84.3384239,359.738885 161.400996,296.055471 249.990733,296.055471 C338.580469,296.055471 416.021447,359.738885 499.990733,359.738885"
            id="路径-2备份-7"
            stroke="url(#linearGradient-2)"
            stroke-dasharray="672"
            stroke-width="1.6"
            transform="translate(249.990733, 327.897178) scale(1, -1) translate(-249.990733, -327.897178) "
          >
            <animate
              attributeName="stroke-dashoffset"
              dur="1.6s"
              repeatCount="indefinite"
              values="-672;0;672"
            />
          </path>
          <path
            d="M-0.00926722543,364.738885 C84.3384239,364.738885 161.400996,301.055471 249.990733,301.055471 C338.580469,301.055471 416.021447,364.738885 499.990733,364.738885"
            id="路径-2备份-8"
            stroke="url(#linearGradient-2)"
            stroke-dasharray="672"
            stroke-width="1.6"
            transform="translate(249.990733, 332.897178) scale(1, -1) translate(-249.990733, -332.897178) "
          >
            <animate
              attributeName="stroke-dashoffset"
              dur="1.6s"
              repeatCount="indefinite"
              values="-672;0;672"
            />
          </path>
          <path
            d="M-0.00926722543,369.738885 C84.3384239,369.738885 161.400996,306.055471 249.990733,306.055471 C338.580469,306.055471 416.021447,369.738885 499.990733,369.738885"
            id="路径-2备份-9"
            stroke="url(#linearGradient-2)"
            stroke-dasharray="672"
            stroke-width="1.6"
            transform="translate(249.990733, 337.897178) scale(1, -1) translate(-249.990733, -337.897178) "
          >
            <animate
              attributeName="stroke-dashoffset"
              dur="1.6s"
              repeatCount="indefinite"
              values="-672;0;672"
            />
          </path>
          <rect
            fill="#049FF7"
            fill-rule="nonzero"
            height="18.5394529"
            id="矩形"
            opacity="1"
            width="147.16999"
            x="176.415005"
            y="238.306353"
          />
        </g>
      </svg>
    </span>
  </div>
</div>
`;
